File: Day2/strong-number.js
function calculateFactorial(n){
    let factorial =1;
    while (n !=0){
        factorial *= n;
        n--;
    }
    return factorial;
}
function isStrongNumber(num){
    let temp = num ,  sum = 0;
    while( num != 0){
        let digit = num % 10;
        sum += calculateFactorial(digit);
        num = Math.floor(num/10);
    }
    return temp == sum;
}
let num = parseInt(prompt("Enter the Number: "));
let res= isStrongNumber(num);
if (res) {
    console.log(num," is a Strong Number.");
} else {
    console.log(num," is not a Strong number.");
}

File: Day2/StrongNumber.java
import java.util.Scanner;
public class StrongNumber {
    
    public static int factorial(int n){

        int factorial=1;
        while(n != 0){
            factorial *= n;
            n--;
        }
        return factorial;
    }
    public static boolean getIsStrongNumber(int num){

        int temp = num;
        int sum =0;
        while(num != 0){
            int digit = num % 10;
            sum += factorial(digit);
            num /= 10;
        }
        return temp == sum;
    }
    public static void main(String[] args) {
    
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the Number: ");
        int num = sc.nextInt();
        sc.close();
        boolean ans = getIsStrongNumber(num);

        if (ans==true) {
            System.out.println(num + "  is a Strong Number.");
        }
        else{
            System.out.println(num + "  is not a Strong Number.");
        }

    }

}
